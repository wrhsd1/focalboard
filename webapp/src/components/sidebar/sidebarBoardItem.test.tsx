// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react'

import {createMemoryHistory} from 'history'
import {Router} from 'react-router-dom'

import {render} from '@testing-library/react'

import {Provider as ReduxProvider} from 'react-redux'

import configureStore from 'redux-mock-store'

import {TestBlockFactory} from '../../test/testBlockFactory'

import {wrapIntl} from '../../testUtils'

import SidebarBoardItem from './sidebarBoardItem'

describe('components/sidebarBoardItem', () => {
    const board = TestBlockFactory.createBoard()

    const view = TestBlockFactory.createBoardView(board)
    view.fields.sortOptions = []
    const history = createMemoryHistory()

    const categoryBoards1 = TestBlockFactory.createCategoryBoards()
    categoryBoards1.name = 'Category 1'
    categoryBoards1.boardIDs = [board.id]

    const categoryBoards2 = TestBlockFactory.createCategoryBoards()
    categoryBoards2.name = 'Category 2'

    const categoryBoards3 = TestBlockFactory.createCategoryBoards()
    categoryBoards3.name = 'Category 3'

    const allCategoryBoards = [
        categoryBoards1,
        categoryBoards2,
        categoryBoards3,
    ]

    const state = {
        users: {
            me: {
                id: 'user_id_1',
            },
        },
        boards: {
            current: board.id,
            boards: {
                [board.id]: board,
            },
        },
        views: {
            current: view.id,
            views: {
                [view.id]: view,
            },
        },
        teams: {
            current: {
                id: 'team-id',
            },
        },
    }

    test('sidebar board item', () => {
        const mockStore = configureStore([])
        const store = mockStore(state)

        const component = wrapIntl(
            <ReduxProvider store={store}>
                <Router history={history}>
                    <SidebarBoardItem
                        categoryBoards={categoryBoards1}
                        board={board}
                        allCategories={allCategoryBoards}
                        isActive={true}
                        showBoard={jest.fn()}
                        showView={jest.fn()}
                        onDeleteRequest={jest.fn()}
                    />
                </Router>
            </ReduxProvider>,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('renders default icon if no custom icon set', () => {
        const mockStore = configureStore([])
        const store = mockStore(state)
        const noIconBoard = { ...board, icon: '' }

        const component = wrapIntl(
            <ReduxProvider store={store}>
                <Router history={history}>
                    <SidebarBoardItem
                        categoryBoards={categoryBoards1}
                        board={noIconBoard}
                        allCategories={allCategoryBoards}
                        isActive={true}
                        showBoard={jest.fn()}
                        showView={jest.fn()}
                        onDeleteRequest={jest.fn()}
                    />
                </Router>
            </ReduxProvider>,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })
})
