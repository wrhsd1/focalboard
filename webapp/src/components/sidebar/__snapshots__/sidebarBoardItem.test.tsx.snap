// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebarBoardItem renders default icon if no custom icon set 1`] = `
<div>
  <div
    class="SidebarBoardItem subitem active"
  >
    <div
      class="octo-sidebar-icon"
    >
      <i
        class="CompassIcon icon-product-boards undefined"
      />
    </div>
    <div
      class="octo-sidebar-title"
      title="board title"
    >
      board title
    </div>
    <div>
      <div
        aria-label="menuwrapper"
        class="MenuWrapper x"
        role="button"
      >
        <button
          class="IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-dots-horizontal OptionsIcon"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="SidebarBoardItem sidebar-view-item active"
  >
    <svg
      class="BoardIcon Icon"
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        opacity="0.8"
      >
        <path
          clip-rule="evenodd"
          d="M4 4H20V20H4V4ZM2 4C2 2.89543 2.89543 2 4 2H20C21.1046 2 22 2.89543 22 4V20C22 21.1046 21.1046 22 20 22H4C2.89543 22 2 21.1046 2 20V4ZM8 6H6V12H8V6ZM11 6H13V16H11V6ZM18 6H16V9H18V6Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </g>
    </svg>
    <div
      class="octo-sidebar-title"
      title="view title"
    >
      view title
    </div>
  </div>
</div>
`;

exports[`components/sidebarBoardItem sidebar board item 1`] = `
<div>
  <div
    class="SidebarBoardItem subitem active"
  >
    <div
      class="octo-sidebar-icon"
    >
      i
    </div>
    <div
      class="octo-sidebar-title"
      title="board title"
    >
      board title
    </div>
    <div>
      <div
        aria-label="menuwrapper"
        class="MenuWrapper x"
        role="button"
      >
        <button
          class="IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-dots-horizontal OptionsIcon"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="SidebarBoardItem sidebar-view-item active"
  >
    <svg
      class="BoardIcon Icon"
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        opacity="0.8"
      >
        <path
          clip-rule="evenodd"
          d="M4 4H20V20H4V4ZM2 4C2 2.89543 2.89543 2 4 2H20C21.1046 2 22 2.89543 22 4V20C22 21.1046 21.1046 22 20 22H4C2.89543 22 2 21.1046 2 20V4ZM8 6H6V12H8V6ZM11 6H13V16H11V6ZM18 6H16V9H18V6Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </g>
    </svg>
    <div
      class="octo-sidebar-title"
      title="view title"
    >
      view title
    </div>
  </div>
</div>
`;
